### PR TITLE
feature/ECCMD-1081_standardize_fields

### PR DESCRIPTION
**Pull request by JavierAlfonsoSantiagoSalazar**
image

### CODE REVIEW

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextCurrencyField.kt [hunk 1]
```diff
@@ -49,14 +49,14 @@ class EditTextCurrencyField(context: Context, attrs: AttributeSet) :
...
-        if (focused) {
-            if (editText?.text.isNullOrEmpty()) {
-                editText?.setText(DOLLAR_SYMBOL)
-            }
-        }
-    }
+//    override fun onFocusChange(v: View?, hasFocus: Boolean) {
+//        super.onFocusChange(v, hasFocus)
+//        if (hasFocus) {
+//            if (editText?.text.isNullOrEmpty()) {
```
Comment (JhoanGamarra) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextCurrencyField.kt:55]: remove commented code

File: test/src/main/res/layout/activity_mock.xml [hunk 2]
```diff
@@ -26,4 +26,11 @@
         app:content="@layout/activity_collapsible_content_view_test"
         app:subtitle="Collapsible subtitle"
         app:collapsibleTitle="Collapsible title"/>
+
+    <co.condorlabs.customcomponents.customedittext.EditTextCurrencyField
+        android:layout_width="match_parent"
```
Comment (JhoanGamarra) [inline test/src/main/res/layout/activity_mock.xml:31]: u should remove that, no?

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextCurrencyField.kt (hunk 1 above)
Comment (JavierAlfonsoSantiagoSalazar) [reply app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextCurrencyField.kt:55]: Done

File: test/src/main/res/layout/activity_mock.xml (hunk 2 above)
Comment (JavierAlfonsoSantiagoSalazar) [reply test/src/main/res/layout/activity_mock.xml:31]: Done

File: app/src/main/java/co/condorlabs/customcomponents/helper/Extensions.kt [hunk 3]
```diff
@@ -16,27 +16,25 @@
...
-    }
-
-    return false
+fun BigDecimal.toDollarAmount(): String {
+    return formatDollarAmount(MONEY_FORMAT)
 }
 
-fun BigDecimal.toDollarAmount(): String {
+fun BigDecimal.toDollarAmountZero(): String {
+    return formatDollarAmount(MONEY_FORMAT_ZERO)
```
Comment (JhoanGamarra) [inline app/src/main/java/co/condorlabs/customcomponents/helper/Extensions.kt:31]: unnecessary , I think, no?

File: test/src/main/java/co/condorlabs/customcomponents/test/MockActivity.kt [hunk 4]
```diff
@@ -24,12 +24,16 @@ import androidx.appcompat.app.AppCompatActivity
  */
 class MockActivity : AppCompatActivity() {
 
+
```
Comment (JhoanGamarra) [inline test/src/main/java/co/condorlabs/customcomponents/test/MockActivity.kt:27]: remove blank line

File: test/src/main/java/co/condorlabs/customcomponents/test/MockActivity.kt [hunk 5]
```diff
@@ -24,12 +24,16 @@ import androidx.appcompat.app.AppCompatActivity
...
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(layout)
+
```
Comment (JhoanGamarra) [inline test/src/main/java/co/condorlabs/customcomponents/test/MockActivity.kt:31]: x2

Comment (JavierAlfonsoSantiagoSalazar) [reply test/src/main/java/co/condorlabs/customcomponents/test/MockActivity.kt]: Done

File: test/src/main/java/co/condorlabs/customcomponents/test/MockActivity.kt (hunk 4 above)
Comment (JavierAlfonsoSantiagoSalazar) [reply test/src/main/java/co/condorlabs/customcomponents/test/MockActivity.kt]: Done

File: app/src/main/java/co/condorlabs/customcomponents/helper/Extensions.kt (hunk 3 above)
Comment (JavierAlfonsoSantiagoSalazar) [reply app/src/main/java/co/condorlabs/customcomponents/helper/Extensions.kt:31]: Done

File: app/src/main/java/co/condorlabs/customcomponents/Constants.kt [hunk 6]
```diff
@@ -84,8 +84,9 @@ const val MAXLINES_DEFAULT = 5
 /**MONEY FORMATS**/
 const val DOLLAR_SYMBOL = "$"
 const val MONEY_FORMAT = "$###,###.##"
```
Comment (leidyzulu) [inline app/src/main/java/co/condorlabs/customcomponents/Constants.kt:86]: we continue using this?

File: app/src/main/java/co/condorlabs/customcomponents/helper/masks/PriceTextWatcherMask.kt [hunk 7]
```diff
@@ -17,131 +17,59 @@
...
-import java.math.RoundingMode
-import java.text.DecimalFormat
 import java.text.NumberFormat
-import java.util.*
 
 /**
  * @author Oscar Gallon on 12/20/18.
  */
-class PriceTextWatcherMask(private val receiver: EditText) : TextWatcherAdapter() {
+class PriceTextWatcherMask(private val receiver: EditText) : TextWatcher {
```
Comment (leidyzulu) [inline app/src/main/java/co/condorlabs/customcomponents/helper/masks/PriceTextWatcherMask.kt:32]: verify in the class which constants can be deleted

File: test/build.gradle [hunk 8]
```diff
@@ -54,6 +54,7 @@ dependencies {
 
     implementation kotlinCoroutines
     implementation kotlinCoroutinesAndroid
+    implementation 'com.google.android.material:material:1.6.1'
```
Comment (leidyzulu) [inline test/build.gradle:57]: continue the behavior of the others implementations

**Review by leidyzulu — CHANGES_REQUESTED** [review]

File: app/src/main/java/co/condorlabs/customcomponents/helper/masks/PriceTextWatcherMask.kt (hunk 7 above)
Comment (JavierAlfonsoSantiagoSalazar) [reply app/src/main/java/co/condorlabs/customcomponents/helper/masks/PriceTextWatcherMask.kt:32]: Done

**Review by leidyzulu — APPROVED** [review]

**Review by JhoanGamarra — APPROVED** [review]

**Pull request merged**